packages/react-components/react-carousel/library/src/components/CarouselCard/useCarouselCard.ts

```diff
@@ -68,10 +69,12 @@ export const useCarouselCard_unstable = (
   const handleFocusCapture = React.useCallback(
     (e: React.FocusEvent) => {
       if (!e.defaultPrevented && isHTMLElement(e.currentTarget) && !isMouseEvent.current) {
-        selectPageByElement(e, e.currentTarget, true);
+        // We want to prevent any browser scroll intervention for 'offscreen' focus
+        containerRef?.current?.scrollTo(0, 0);
```

Comment:
Please check if this works with screen readers i.e. VoiceOver, I had bad luck in a similar case recently.